pkgs/development/tools/rust/cargo-llvm-cov/default.nix

```diff
@@ -0,0 +1,24 @@
+{ lib, rustPlatform, fetchCrate, libllvm }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "cargo-llvm-cov";
+  version = "0.4.13";
+
+  doCheck = false;
+
+  buildInputs = [ libllvm ];
+
+  src = fetchCrate {
+    inherit pname version;
+    sha256 = "sha256-x1M1rvQXiNX1HUgBtCLJy2Pdkes8I0ucGpNSweT5Tfs=";
+  };
+
+  cargoSha256 = "sha256-IOngZiyONbxPzpNdKE6UM9hlZcV4ZfKZdVPX9+vgqjI=";
+
+  meta = with lib; {
+    description = "Cargo llvm cov generates code coverage via llvm.";
```

Comment:
To fix running the tests and to fix running `cargo llvm-cov` without a rustup toolchain installation:
```suggestion
{ lib, rustPlatform, fetchCrate, fetchFromGitHub, libllvm, makeBinaryWrapper }:

rustPlatform.buildRustPackage rec {
  pname = "cargo-llvm-cov";
  version = "0.4.13";

  # required because upstream does not keep a Cargo.lock file in their repository
  # (and some of the files required for running the tests are not published to crates.io)
  src = fetchFromGitHub {
    owner = "taiki-e";
    repo = "cargo-llvm-cov";
    rev = "v${version}";
    sha256 = "sha256-DfkBBPlJhxKeH6LZPAm3AQfqwNWTTSlxANUxMTGha28=";
  };
  srcCrate = fetchCrate {
    inherit pname version;
    sha256 = "sha256-x1M1rvQXiNX1HUgBtCLJy2Pdkes8I0ucGpNSweT5Tfs=";
  };
  prePatch = ''
    cp ${srcCrate}/Cargo.lock .
  '';

  nativeBuildInputs = [ makeBinaryWrapper ];
  buildInputs = [ libllvm ];

  LLVM_COV = "${rustPlatform.rust.rustc.llvmPackages.llvm}/bin/llvm-cov";
  LLVM_PROFDATA = "${rustPlatform.rust.rustc.llvmPackages.llvm}/bin/llvm-profdata"; 

  postInstall = ''
    wrapProgram $out/bin/cargo-llvm-cov \
      --set LLVM_COV "${rustPlatform.rust.rustc.llvmPackages.llvm}/bin/llvm-cov" \
      --set LLVM_PROFDATA "${rustPlatform.rust.rustc.llvmPackages.llvm}/bin/llvm-profdata"
  '';

  cargoSha256 = "sha256-IOngZiyONbxPzpNdKE6UM9hlZcV4ZfKZdVPX9+vgqjI=";

  meta = with lib; {
    description = "Cargo subcommand to gather source-based code coverage using LLVM";
```
(may not be the most elegant solution)